chevah/compat
# [#3228] Strict new file perms. Non-world readable PID file.

Scope
=====

This updates compat so that new files have non word readable/writable permission by default.

The previous default behaviour was to use 666 + UMASK.
But in some cases, there was no umaks, of the umask didn't had a good value.


Changes
=======

Add a mode argument when opening file operations which might create a new file.

As a cleanup, I have remove the UTF-8 magic handling.
I have also removed the openforUpdating as I don't know what is that and is not used.


How to try and test the changes
===============================

reviewers: @dumol 

Check that changes make sense.

I will create a separate PR for chevah/server to test the actual daemon permissions.

For now, I have created a demo daemon script which can be used for manual testing the changes

I was testing it as

```
$ paver build
$ ./build-ubuntu2004-x64/bin/python example/unix-daemon.py 
$ cat /tmp/py-daemon-example.log
Starting in fork...
1588267432.14 Still alive.
1588267433.14 Still alive.
$ cat /tmp/py-daemon-example.pid 
560687
$ kill 560687
$ cat /tmp/py-daemon-example.pid 
cat: /tmp/py-daemon-example.pid: No such file or directory
```
